test/unit/submission/validation.spec.ts

```diff
@@ -3134,6 +3134,179 @@ describe('data-validator', () => {
 			chai.expect(result[ClinicalEntitySchemaNames.SURGERY].dataErrors[0]).to.deep.eq(error);
 		});
 
+		it('should error when submitted reference_radiation_treatment_id does not match submitted treatment_id', async () => {
```

Comment:
I added test cases to test against current Donor, submitted Donor, and Donor in DB